pkg/fleet/daemon/daemon.go

```diff
@@ -185,6 +190,10 @@ func (d *daemonImpl) Start(_ context.Context) error {
 				if err != nil {
 					log.Errorf("Daemon: could not run GC: %v", err)
 				}
+			case <-time.After(refreshStateInterval):
```

Comment:
is refersh state interval is shorter than gcInterval it will never do garbage collection and vice versa